packages/firestore/src/api/database.ts

```diff
@@ -296,6 +316,32 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
       'FirestoreSettings.host cannot be falsey'
     );
 
+    if (!this._config.settings.timestampsInSnapshots) {
+      log.error(`
+The behavior for Date objects stored in Firestore is going to change
+AND YOUR APP MAY BREAK.
+To hide this warning and ensure your app does not break, you need to add the
+following code to your app before calling any other Cloud Firestore methods:
+
+  const firestore = firebase.firestore(firebase.app());
```

Comment:
I think `firebase.app()` is not necessary?  Just doing `firebase.firestore()` should use the default app...

Reply:
So it's a bit convoluted.  firebase.firestore() is actually a function that FirebaseApp creates when we (Firestore) call firebase.INTERNAL.registerService() to register ourselves.  So:

1. We call firebase.INTERNAL.registerService() here: https://github.com/firebase/firebase-js-sdk/blob/b5aa5b8335cd13ce8c30c188843723f3a766ac53/packages/firestore/src/platform/config.ts#L58
2. It creates this `serviceNamespace` which is the firebase.firestore() function [see line 369] except it also has properties like firebase.firestore.FieldValue hence it's called a "namespace": https://github.com/firebase/firebase-js-sdk/blob/b5aa5b8335cd13ce8c30c188843723f3a766ac53/packages/app/src/firebaseApp.ts#L353
3. That serviceNamespace function defaults to using the default app (`appArg: FirebaseApp = app()`).

So app is optional...

Reply:
Thanks! I tried finding it in the code, realized it's something dynamically created and gave up.